### DISCUSSION TITLE
Galaxy S4 SPH-L720 (jfltespr) (Sprint) GSM Use not available!

### DISCUSSION THREAD

**Discussion by IzumiSenaSora** [Divested-Mobile/DivestOS-Build#367] [Q&A]
Hi i bought a Galaxy S4 (it was a dream phone in past of mine).

I got a Galaxy S4 which on back it says it was GT-i9505!
But in bootloader it says SPH-L720!
Then i tried to flash GT-i9505 stock rom it failed!
Then i tried SPH-L720 it worked!

Now after installing DivestOS 11 no GSM Sim use available!
But i am outside of US so no CDMA network here!
After inserting SIM it says my SIM Card name but No Service!

Can someone help me how i can use my SIM outside US with GSM Network?

**Comment by SkewedZeppelin** [comment]
The device might be locked to Sprint by IMEI.

**Reply by IzumiSenaSora** [reply]
I unlocked it!
Finally!
Then updated to stock lollipop firmware!
It worked fine!
But after installing DivestOS it shows "No services" what can i do?

**Reply by IzumiSenaSora** [reply]
After so many attempt i finally fixed it.

But little bit different way.

I wanted to check if it really "jfltespr" build problem thats why i tried to install i9505G (Google Play Edition) "boot, recovery, system" after boot SIM worked.

Then i though why not install "jfltexx" instead of "jfltespr" DivestOS build.

I tried to install, but it failed. Cause devices codename is different.

I modified the "updater-script" file to allow "jfltespr" in "jfltexx" build zip file.

Then installed it. After installed, then factory reset from recovery then boot.

And again after boot SIM worked!

So in Summary i installed "jfltexx (i9505)" build in "jfltespr (L720)" Phone.

Now why am i sharing this to you?

Cause you know better then anyone if it's ok to do this?

Or should i try to find some other way from "jfltespr (L720)" build? Instead of relying on "jfltexx (i9505)" build.

Kindly share your thoughts please. If there's any risk or anything, is it safe or not

**Reply by SkewedZeppelin** [reply]
glad to hear you found a fix.
if it works then stick with it!

**Reply by IzumiSenaSora** [reply]
It work's great!

In jfltespr build (DivestOS/LineageOS) i get some screen issues like this!
![galaxy-s4-running-lineageos-18-sporadically-reboots-then-v0-g1v7s8vdggva1](https://github.com/user-attachments/assets/0d4e0faa-ec83-496a-b01a-03e0829ac0e8)

It get stuck and then turn off then turn on again but in jfltexx build no issue like this!

Only issue i have is i can't update it via updater app.

Cause i need to change updater-script to allow jfltespr in jfltexx build file.

Can you allow jfltexx build to be able to install in jfltespr via updater app?

**Reply by SkewedZeppelin** [reply]
if you manually flashed the jfltexx recovery it should continue to allow you to install jfltexx builds without any edits

**Reply by IzumiSenaSora** [reply]
I manually flashed DivestOS JFLTE XX Recovery

![Screenshot_20241130-154352_Files](https://github.com/user-attachments/assets/dbba58c2-8ec0-4ffb-bd29-61d6119c2af7)

Then rebooted into DivestOS Recovery i tried to install DivestOS JFLTE XX (Unmodified) Firmware then it failed! Saying my device is JFLTE SPR

Then i entered in fastboot mode in DivestOS Recovery and found this!

![IMG_20241130_154421](https://github.com/user-attachments/assets/ca4d65b6-2365-4827-82cc-9f7951f68510)

It says JFLTE SPR even though i installed JFLTE XX Recovery.

**Reply by SkewedZeppelin** [reply]
that is very weird, I don't understand why it would do that

**Reply by IzumiSenaSora** [reply]
Maybe properties get generated while installation from bootloader information instead of hardcoded in a file?

Edit: I tried to update it via "Settings/System Updater" after reboot into recovery it shows "Error!" Then rebooted into system.

That means even if i try to update via Settings it still checks if it JFLTE XX or not.

**Reply by IzumiSenaSora** [reply]
Can you add JFLTE SPR support in "updater-script" file in JFLTE XX build please

**Reply by SkewedZeppelin** [reply]
I don't think this can be easily done.

**Reply by IzumiSenaSora** [reply]
How about adding `jfltespr` in `TARGET_OTA_ASSERT_DEVICE` Variable [LineageOS Repo Link](https://github.com/LineageOS/android_device_samsung_jfltexx/blob/221635b6daa1f162b8c18350bd16b15b8c8f616e/BoardConfig.mk#L23)

```
TARGET_OTA_ASSERT_DEVICE := jflte,jfltecan,jfltetmo,jfltexx,jgedlte
```
Replace it To this via `sed` command
```
TARGET_OTA_ASSERT_DEVICE := jflte,jfltecan,jfltespr,jfltetmo,jfltexx,jgedlte
```
